htmresearch/frameworks/nlp/classification_model.py

```diff
@@ -371,3 +383,103 @@ def classifyText(self, token, reset=0):
                            implies no decision could be made.
     """
     raise NotImplementedError
+
+
+  def save(self, saveModelDir):
+    """ Save the model in the given directory via pickling.
+
+    @param saveModelDir (string)
+           Directory path for saving the model. This directory should
+           only be used to store a saved model. If the directory does not exist,
+           it will be created automatically and populated with model data. A
+           pre-existing directory will only be accepted if it contains previously
+           saved model data. If such a directory is given, the full contents of
+           the directory will be deleted and replaced with current model data.
+
+    Implementation note: Subclasses should override _serializeExtraData() to
+    save any data that cannot be pickled.
+    """
+    saveModelDir = os.path.abspath(saveModelDir)
+    modelPickleFilePath = os.path.join(saveModelDir, "model.pkl")
+
+    # Delete old model directory if we detect it
+    if os.path.exists(saveModelDir):
+      if not os.path.isdir(saveModelDir) or \
+         not os.path.isfile(modelPickleFilePath):
+        raise Exception(("Existing filesystem entry <%s> is not a model"
+                         " checkpoint -- refusing to delete"\
+                         " (%s missing or not a file)") % \
+                          (saveModelDir, modelPickleFilePath))
+
+      shutil.rmtree(saveModelDir)
+
+    # Create a new checkpoint directory for saving state
+    self.__makeDirectoryFromAbsolutePath(saveModelDir)
+
+    with open(modelPickleFilePath, "wb") as modelPickleFile:
+      pkl.dump(self, modelPickleFile)
+
+    # Tell the model to save extra data, if any.
+    self._serializeExtraData(saveModelDir)
+
+
+  @classmethod
+  def load(cls, savedModelDir):
+    """Create model from saved checkpoint and return it.
```

Comment:
> Please add that it is expected the model is named "model.pkl".

That's an internal implementation detail - I don't think that should be in the user facing comments.  From the user's standpoint the saved model is the entire directory.